lib/kubernetes-deploy/kubernetes_resource/daemon_set.rb

```diff
@@ -17,8 +18,15 @@ def status
 
     def deploy_succeeded?
       return false unless exists?
-      rollout_data["desiredNumberScheduled"].to_i == rollout_data["updatedNumberScheduled"].to_i &&
-      rollout_data["desiredNumberScheduled"].to_i == rollout_data["numberReady"].to_i &&
+
+      puts pods.inspect
+      puts "TOTAL_PODS #{@pods.size}"
+      considered_pods = @pods.select { |p| @nodes.map(&:name).include?(p.node_name) }
```

Comment:
now that I think about this more, this check should probably be gated by `rollout_data["desiredNumberScheduled"].to_i == rollout_data["updatedNumberScheduled"].to_i`